Improved whisper menu

Mitchdev opened the pull request:
Show all your whispers and indicate if the user is online

![image](https://user-images.githubusercontent.com/13901269/232690079-2908d7cd-a00e-4a15-8790-f357e174ce8d.png)

Mitchdev committed:
Improved whisper menu

Mitchdev committed:
Remove test whipser

11k reviewed (changes requested):
I'd like to finally take a look at this. 😅 Would you mind merging in master and fixing any merge conflicts?

Mitchdev committed:
Merge branch 'master' into whispers-menu

Mitchdev committed:
Simplify online/offline and tidy

Mitchdev committed:
Fix prettier problems

Mitchdev commented:
For some reason the prettier commit didn't merge when I merged the master branch in.

11k reviewed (changes requested):
Would you mind fixing the merge conflicts, and humanizing the date next to each conversation (e.g., "a month ago")? You can give the date a title so that it displays a local time in a localized format on hover.

Mitchdev committed:
Merge branch 'master' into whispers-menu

Mitchdev committed:
Use relative time.

11k commented on assets/chat/js/menus/ChatWhisperUsers.js:
```diff
@@ -46,31 +70,37 @@ export default class ChatWhisperUsers extends ChatMenu {
+      <span class="user">${whisper.nick}</span>
+      <div class="right">
+        ${unread}
+        <span class="time" title="${time.format(
+          'LLL',
+        )}">${time.fromNow()}</span>
```
Would you mind using code tags to make this a little more readable? It's a dependency we added sometime after this PR.

11k commented on assets/chat/css/style.scss:
```diff
@@ -1333,66 +1333,84 @@ hr {
+        width: 8px;
+        height: 8px;
+        margin-right: $gutter-sm;
+        -moz-border-radius: 7.5px;
+        -webkit-border-radius: 7.5px;
```
Are vendor prefixes still necessary here?

11k commented on assets/chat/css/style.scss:
```diff
@@ -1333,66 +1333,84 @@ hr {
-      color: $color-accent-light;
-      .badge {
-        color: $color-chat-text1;
+  .section {
```
I find this CSS challenging to read and I think it's a combination of there being no newlines before each nested ruleset and excessive nesting. For instance, you don't need this level of specificity to select `.unread`.
```
#chat-whisper-users .section .user-entry .right .unread {
}
```
I think the issue is you're trying to make the nesting in your CSS reflect the nesting in your markup. Besides readability, you can also run into situations where you have to rely on `!important` more often to override styles.

Mitchdev committed:
Merge branch 'master' into whispers-menu

Mitchdev committed:
Update scss

Mitchdev committed:
Update and refactor js

11k committed:
Improve markup and styles

Rework CSS to (loosely) follow BEM guidelines for better readability and
to reduce nesting. Simplify CSS by removing unneeded rules and using
logical margin and padding properties. Simplify conversation markup by
relying on flexbox to correctly position right-aligned elements.

Convert online/offline indicator from a pseudo-element to a standard
element because the latter is more appropriate. Render the timestamp of
conversations as a `time` element because it's semantically correct.

Fix layout issue that occurs between username, unread, and time elements
when chat font scaling is increased.
